ctapipe/image/extractor.py

```diff
@@ -227,8 +235,31 @@ def integration_correction(
     return correction
 
 
+def slide_window(waveform, width):
+    """Smooth a pixel's waveform (or a slice of it) with a kernel of certain
+     size via convolution.
+
+    Parameters
+    ----------
+    waveform : array_like
+        DL0-level waveform (or slice of it) of one event.
+        Shape: max (n_samples)
+    width : int
+        Size of the smoothing kernel.
+
+    Returns
+    -------
+    sum : array_like
+        Array containing the sums for each of the kernel positions.
+        Shape: max (n_samples - (window_width - 1))
+
+    """
+    sums = np.convolve(waveform, np.ones(width, dtype=int), "valid")
+    return sums
+
+
 class ImageExtractor(Component):
-    def __init__(self, subarray, config=None, parent=None, **kwargs):
```

Comment:
Remove this change, subarray should be a required parameter. I presume this was an accident in merging the recent PRs